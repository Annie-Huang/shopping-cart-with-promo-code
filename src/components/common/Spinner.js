// Problem #1: No loading indicator on course list
// Problem #2: No loading indicator when saving a course
// Problem #3: No loading indicator when loading a course
// Solution: Display a spinner while loading

import React from "react";
import "./Spinner.css";

const Spinner = () => {
    return <div className="loader">Loading...</div>;
};

export default Spinner;
