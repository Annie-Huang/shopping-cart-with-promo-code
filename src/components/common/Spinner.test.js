import React from "react";
import toJson from "enzyme-to-json";
import {shallow} from "enzyme/build";
import Spinner from "./Spinner";

describe('Spinner presentation', () => {
    it('should render Spinner component', () => {
        expect(toJson(shallow(<Spinner />))).toMatchSnapshot();
    });
});
