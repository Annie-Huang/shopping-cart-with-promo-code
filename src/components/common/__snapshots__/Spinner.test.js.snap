// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner presentation should render Spinner component 1`] = `
<div
  className="loader"
>
  Loading...
</div>
`;
